### ISSUE TITLE
Nested extractions not possible

### ISSUE THREAD

**Issue by kisenka** [webpack-contrib/extract-text-webpack-plugin#470]
tl;dr: currently it is not possible to extract any resource (e.g. svg) from another extractable resource (e.g. css). For example following stylesheet should be extracted:

```css
.image { background-image: url('./image.svg'); }
```

It's ok, but if there is second extract plugin instance configured for SVGs - no SVGs will be extracted and stylesheet code will be compiled to:

```css
.image { background-image: url([object Object]); }
```

Is there a workaround for this?
Repo with reproducible behaviour - https://github.com/kisenka/multiple-extract-text-webpack-plugins. Installation:

```
npm install
npm run build
```

## Description

I want to do the following thing:

1. Merge many SVGs in a single sprite (done via svg-sprite-loader).
2. Extract sprite as separate file (done via svg-sprite-loader/lib/extract-svg-plugin, tiny wrapper over the extract-text-webpack-plugin which just wraps result with <svg> tags to get valid svg markup).
3. Return path to extracted SVG with fragment identifier which refers to specific sprite symbol, e.g. `sprite.svg#image`. This is doable unless you want to refer SVG image (which is extracted via extract plugin) in CSS which is also extracted via second extract plugin instance.

Example:

**webpack.config.js**

```js
var TextExtractPlugin = require('extract-text-webpack-plugin');

var CSSExtractor = new TextExtractPlugin({ filename: '[name].css' });
var SVGExtractor = new TextExtractPlugin({ filename: '[name].svg' });

module: {
  rules: [
    {
      test: /\.css$/,
      use: CSSExtractor.extract({ use: 'css-loader' })
    },
    
    // emulate svg-sprite-loader for imports from CSS
    {
      test: /\.svg$/,
      use: SVGExtractor.extract({ use: 'raw-loader' })
    }
  ]
}
```

**styles.css**

```css
.img {
  background-image: url('./image.svg');
}
```

CSS will be compiled to

```css
.img {
  background-image: url([object Object]);
}
```

I think it's because in first (SVG) extract compilation all image source was removed by extract-plugin, and in the second (CSS) extract compilation gets cached module result from webpack with empty content. I am trying to patch extract plugin loader and return some result instead of `// // removed by extract-text-webpack-plugin`, but have no success. In second extract compilation image module still has empty content.
Is it possible to implement desired behaviour with extract plugin?

**Comment by evilebottnawi** [comment]
@kisenka why don't use `file-loader` (`url-loader` or other) fro this?

**Comment by kisenka** [comment]
@evilebottnawi because I want to ExtractPlugin to do it - merge all modules in one chunk. But I solve this via custom plugin, so the issue can be closed.

**Comment by evilebottnawi** [comment]
@kisenka `extract-text-webpack-plugin` don't know how to merge `svg` and other types, it is just remove content from `js` to other chunk. But can your provide link to custom plugin, maybe we can found solution, thanks!

**Comment by kisenka** [comment]
@evilebottnawi you're right, that's why I wrote a plugin which extends ExtractPlugin and do just one thing - wraps extracted result with `<svg></svg>` :) But it was made with 1.x version of ExtractPlugin, [here is a source](https://github.com/kisenka/svg-sprite-loader/blob/e0cfbbb380232b03d450bc634bde0583ea07bcf9/lib/extract-svg-plugin.js).

**Comment by kisenka** [comment]
@evilebottnawi if you mind I can make a PR to improve ExtractPlugin processing of SVGs.

**Comment by kisenka** [comment]
But still you can't get it works with double extraction - extract SVGs from extracting CSS. That's why I handle this by separate plugin which emit another chunk with extracted SVGs and replace path to it in CSS - https://github.com/kisenka/svg-sprite-loader/blob/master/lib/plugin.js#L73

**Comment by evilebottnawi** [comment]
@kisenka maybe best solution is add event to `extract-text-webpack-plugin` and separate this logic to other plugin (loader), it is allow to handle other types not only svg. Handle svg is out of scope `extract-text-webpack-plugin`, but with PR

**Comment by kisenka** [comment]
@evilebottnawi sounds more elegant. Related issues #85 and #167.

**Comment by michael-ciniawsky** [comment]
ETWP entered maintenance mode and won't receive any feature updates, please open an issue in [`mini-css-extract-plugin`](https://github.com/webpack-contrib/mini-css-extract-plugin) instead
